apps/system/test/marionette/lib/system.js

```diff
@@ -300,6 +307,16 @@ System.prototype = {
     return this.client.findElement(System.Selector.utilityTrayMotion);
   },
 
+  get toaster() {
+    var toaster;
+    this.client.waitFor(function() {
+      toaster = this.client.findElement(System.Selector.toaster);
+      return toaster ? true : false;
+    }.bind(this));
```

Comment:
Ahhh, I didn't realize this didn't do a `waitForElement(System.Selector.toaster)`. If you changed it to do that instead, you could simplify the logic above that waits for it to appear and then disappear.